main/src/library/Core/Io/InputStream.flix

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright 2020 Rasmus Larsen
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+use Core/Io/IOError.IOError;
+
+namespace Core/Io/InputStream {
+    pub opaque type InputStream = ##java.io.InputStream
+
+    pub def open(p: Path): Result[InputStream, IOError] & Impure = try {
+        import new java.io.FileInputStream(##java.io.File) as newFileInputStream;
+        let Path(javaFile) = p;
+        let inputStream = newFileInputStream(javaFile);
+        Ok(InputStream(inputStream as ##java.io.InputStream))
+    } catch {
+        case _: ##java.io.FileNotFoundException => Err(IOError.FileNotFound(p))
+    }
+
+    ///
+    /// Creates an `InputStream` from `b`.
+    ///
+    pub def fromBuffer(b: Array[Int8]): InputStream & Impure = {
+        import new java.io.ByteArrayInputStream(Array[Int8]) as newByteArrayInputStream;
+        let byteStream = newByteArrayInputStream(b);
+        InputStream(byteStream as ##java.io.InputStream)
+    }
+
+    ///
+    /// Returns the number of bytes available in `s`.
+    ///
+    pub def available(s: InputStream): Result[Int32, IOError] & Impure = try {
+        import java.io.InputStream.available();
+        let InputStream(javaStream) = s;
+        Ok(javaStream.available())
+    } catch {
+        case _: ##java.io.IOException => Err(IOError.Generic("I/O Error in Core/Io/InputStream.available"))
+    }
+
+    ///
+    /// Reads a single byte from `s`. If the end of the stream is reached `-1` is returned.
+    ///
+    pub def readByte(s: InputStream): Result[Int32, IOError] & Impure = try {
```

Comment:
The reason is that the byte is returned in the [0-255] range. Along with EOF being reported as `-1` the requirements is that it's an integer strictly larger than an `Int8`. Since the Java API uses their 32-bit integer I decided to keep the API consistent.

Another option could be reading the returned integer and making `readByte` return `Result[Option[Int8], IOError]` where `Ok(None)` is EOF.

Reply:
Ok; that makes sense. I think we leave it as is.